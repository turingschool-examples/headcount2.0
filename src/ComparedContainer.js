import React from 'react';
import ComparedCard from './ComparedCard';
import PropType from 'prop-types';

const ComparedContainer = ({ compareObject }) => {
  return <ComparedCard 
    compareObject={compareObject}
  />;
};

ComparedContainer.propTypes = {
  compareObject: PropType.object
};

export default ComparedContainer;