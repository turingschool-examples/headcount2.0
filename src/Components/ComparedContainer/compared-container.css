.compared-container {
  display: flex;
  justify-content: center;
  width: 100%;
}

.compared-container :nth-child(1) {
  order: 1;
}

.compared-container :nth-child(2) {
  order: 3;
}

.compared-container :nth-child(3) {
  order: 2;
}