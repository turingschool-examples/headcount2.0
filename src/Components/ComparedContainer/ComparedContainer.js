import React from 'react';
import './compared-container.css';
import Card from '../Card/Card';
import CompareCard from '../CompareCard/CompareCard';
import PropTypes from 'prop-types';

const ComparedContainer = ({selectedCards, selectCard, districtMethods}) => {
  const displaySelected = selectedCards.map((card, index) => {
    return <Card {...card} selectCard={selectCard} key={index} />
  })
  
  return (
    <div className="compared-container" >
      {displaySelected}
      <CompareCard districtMethods={districtMethods} selectedCards={selectedCards} className="compare-card" />
    </div>
  )
}

ComparedContainer.propTypes = {
  selectCard: PropTypes.func.isRequired,
  districtMethods: PropTypes.object.isRequired,
  selectedCards: PropTypes.arrayOf(PropTypes.object)
}

export default ComparedContainer;