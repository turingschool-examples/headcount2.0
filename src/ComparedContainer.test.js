import React from 'react';
import { shallow } from 'enzyme';
import ComparedContainer from './ComparedContainer';

describe('ComparedContainer', () => {
  it('should render with all of the appropriate elements', () => {
    const wrapper = shallow(
      console.log(comparedObject)
      <ComparedContainer 
        compareObject={[]}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});