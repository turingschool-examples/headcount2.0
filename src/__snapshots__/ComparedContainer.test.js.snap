// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparedContainer should render with all of the appropriate elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparedContainer
    comparedObject={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "compareObject": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "compareObject": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
